lib/routes/syosetu/ranking.ts

```diff
@@ -0,0 +1,425 @@
+| --- | --- |
+| total | All Works |
+| t | Short Stories |
+| r | Ongoing Series |
+| er | Completed Series |
+
+For \`isekai\` ranking type:
+When multiple works have the same points, their order may differ from syosetu's ranking as syosetu randomizes the order for works with identical points.
+異世界転生/転移ランキングについて：
+集計の結果、同じポイントの作品が複数存在する場合、Syosetu ではランダムで順位が決定されるため、本 Route の順位と異なる場合があります。
+`,
+    radar: [
+        {
+            source: ['yomou.syosetu.com/rank/list/type/:type'],
+            target: '/ranking/list/:type',
+        },
+        {
+            source: ['yomou.syosetu.com/rank/genrelist/type/:type'],
+            target: '/ranking/genre/:type',
+        },
+        {
+            source: ['yomou.syosetu.com/rank/isekailist/type/:type'],
+            target: '/ranking/isekai/:type',
+        },
+        ...getBest5RadarItems(),
+    ],
+};
+
+function parseGeneralRankingType(type: string): { period: RankingPeriod; novelType: NovelType } {
+    const [periodStr, novelTypeStr] = type.split('_');
+
+    const period = periodStr as RankingPeriod;
+    const novelType = novelTypeStr as NovelType;
+
+    if (!Object.values(RankingPeriod).includes(period) || !Object.values(NovelType).includes(novelType)) {
+        throw new InvalidParameterError(`Invalid general ranking type: ${type}`);
+    }
+
+    return { period, novelType };
+}
+
+function parseGenreRankingType(type: string): { period: RankingPeriod; genre: number; novelType: NovelType } {
+    const [periodStr, genreStr, novelTypeStr = NovelType.TOTAL] = type.split('_');
+
+    const period = periodStr as RankingPeriod;
+    const genre = Number(genreStr) as Genre;
+    const novelType = novelTypeStr as NovelType;
+
+    if (!Object.values(RankingPeriod).includes(period) || !Object.values(Genre).includes(genre) || genre === 9904 || genre === 9801 || !Object.values(NovelType).includes(novelType)) {
+        throw new InvalidParameterError(`Invalid genre ranking type: ${type}`);
+    }
+
+    return { period, genre, novelType };
+}
+
+function parseIsekaiRankingType(type: string): { period: RankingPeriod; category: IsekaiCategory; novelType: NovelType } {
+    const [periodStr, categoryStr, novelTypeStr = NovelType.TOTAL] = type.split('_');
+
+    const period = periodStr as RankingPeriod;
+    const category = categoryStr as IsekaiCategory;
+    const novelType = novelTypeStr as NovelType;
+
+    if (!Object.values(RankingPeriod).includes(period) || !Object.values(IsekaiCategory).includes(category) || !Object.values(NovelType).includes(novelType)) {
+        throw new InvalidParameterError(`Invalid isekai ranking type: ${type}`);
+    }
+
+    return { period, category, novelType };
+}
+
+async function handler(ctx: Context): Promise<Data> {
+    const { listType, type } = ctx.req.param();
+    const rankingType = listType as RankingType;
+    const routeParams = querystring.parse(ctx.req.param('routeParams'));
+    const limit = Number(routeParams.limit || 300);
```

Comment:
Please make use of the [common parameter limit](https://docs.rsshub.app/guide/parameters#limit-entries) `ctx.req.query('limit')` instead.